File: src/com/lushprojects/circuitjs1/client/EditTransistorModelDialog.java
package com.lushprojects.circuitjs1.client;

public class EditTransistorModelDialog extends EditDialog {

    TransistorModel model;
    TransistorElm transistorElm;
    
    public EditTransistorModelDialog(TransistorModel dm, CirSim f, TransistorElm te) {
	super(dm, f);
	model = dm;
        transistorElm = te;
	applyButton.removeFromParent();
    }

    boolean apply() {
	super.apply();
//	if (model.name == null || model.name.length() == 0)
//	    model.pickName();
	if (transistorElm != null)
	    transistorElm.newModelCreated(model);
	return true;
    }
    
    public void closeDialog() {
	super.closeDialog();
	EditDialog edlg = CirSim.editDialog;
	CirSim.console("resetting dialog " + edlg);
	if (edlg != null)
	    edlg.resetDialog();	
	CirSim.diodeModelEditDialog = null;
    }
}

File: EditTransistorModelDialog.ts
/*
    Copyright (C) Paul Falstad and Iain Sharp

    This file is part of CircuitJS1.

    CircuitJS1 is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 2 of the License, or
    (at your option) any later version.

    CircuitJS1 is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with CircuitJS1.  If not, see <http://www.gnu.org/licenses/>.
*/

import { EditDialog } from "./EditDialog";
import { CirSim } from "./CirSim";
import { TransistorModel } from "./TransistorModel";
import type { TransistorElm } from "./TransistorElm";

export class EditTransistorModelDialog extends EditDialog {
    model: TransistorModel;
    transistorElm: TransistorElm | null;

    constructor(dm: TransistorModel, f: CirSim, te: TransistorElm | null) {
        super(dm, f);
        this.model = dm;
        this.transistorElm = te;
        this.applyButton.remove();
    }

    apply(): boolean {
        if (!super.apply())
            return false;
        if (this.model.name === null || this.model.name.length === 0)
            this.model.pickName();
        if (this.transistorElm !== null)
            this.transistorElm.newModelCreated(this.model);
        return true;
    }

    closeDialog(): void {
        super.closeDialog();
        const edlg = CirSim.editDialog;
        CirSim.console("resetting dialog " + edlg);
        if (edlg !== null)
            edlg.resetDialog();
        // matches upstream Java, which also clears diodeModelEditDialog here rather than
        // transistorModelEditDialog (see EditTransistorModelDialog.java)
        CirSim.diodeModelEditDialog = null;
    }
}
